client/scala/src/main/scala/ai/verta/repository/Commit.scala

```diff
@@ -338,4 +338,78 @@ class Commit(
+   *  @param walker the FolderWalker to process the folders in the folder tree.
+   *  @return a stream of Try's of WalkOutput.
+   */
+  def walk[T](walker: FolderWalker[T])(implicit ec: ExecutionContext): Stream[Try[T]] = {
+    if (!saved)
+      Stream(Failure(new IllegalCommitSavedStateException("Commit must be saved before it can be walked")))
+    else
+      continueWalk(getFolder(PathList(Nil)).get, walker)
+  }
+
+  /** Get the folder corresponding to a path in list form
+   *  @param location location
+   *  @return the folder, if succeeds
+   */
+  def getFolder(location: PathList)(implicit ec: ExecutionContext): Try[Folder] = {
+    clientSet.versioningService.GetCommitComponent2(
+      commit_sha = id.get,
+      repository_id_repo_id = repo.id,
+      location = if (location.components.length > 0) Some(location.components) else None
+    ).flatMap(r => {
+        val folderPath = location.path
+        val responseFolder = r.folder
+
+        val folderNames = responseFolder
+        .flatMap(_.sub_folders) // Option[List[VersioningFolderElement]]
+        .map(_.map(folder => folder.element_name.get))
+        // Option[List[String]]
+        .map(_.sortWith((name1: String, name2: String) => name1 < name2))
+
+        val blobNames = responseFolder
+        .flatMap(_.blobs) // Option[List[VersioningFolderElement]]
+        .map(_.map(folder => folder.element_name.get))
+        // Option[List[String]]
+        .map(_.sortWith((name1: String, name2: String) => name1 < name2))
+
+        Success(Folder(folderPath, blobNames.getOrElse(Nil), folderNames.getOrElse(Nil)))
+      })
+  }
+
+  /** Continue the walk from a folder.
+   *  @param folder current folder being explored.
+   *  @param walker a FolderWalker instance to process the returned folder
+   *  @return Stream of Trys. If the returned WalkOutput fails, abort the remaining locations.
+   */
+  def continueWalk[T](folder: Folder, walker: FolderWalker[T])(implicit ec: ExecutionContext): Stream[Try[T]] = {
+    // process the folder and walker:
+    val replacedWalker = walker.replace(folder)
+    val filteredFolder = replacedWalker.filterFolder(folder)
+
+    val subfolders: Try[List[Folder]] =
+      Try(filteredFolder.subfolderPaths.map(folder => getFolder(PathList(folder.split("/").toList))).map(_.get))
+    val blobs: Try[List[Blob]] =
+      Try(filteredFolder.blobPaths.map(get).map(_.get))
+
+    subfolders match {
+      case Failure(e) => Stream(Failure(e))
+      case Success(sf) => {
+        blobs match {
+          case Failure(e) => Stream(Failure(e))
+          case Success(bl) => {
+            val blobResults: Stream[Try[T]] =
+              filteredFolder.blobs.zip(bl).toStream.map(pair => Success(replacedWalker.visitBlob(pair._1, pair._2)))
+            val subfolderResults: Stream[Try[T]] =
+              filteredFolder.subfolders.zip(sf).toStream.map(pair => Success(replacedWalker.visitFolder(pair._1, pair._2)))
+
```

Comment:
I recommend to generally avoiding "_" in compound functions like this. It's not clear if "_" will be the argument from the first or second function (e.g. if continue walk took a function with no arguments and returned a function with one argument, this line would look the same or very similar).

Reply:
I believe you meant to comment on the line below (the one with continueWalk function call). Anyway should be fixed.